### PR TITLE
Added an action to deploy a serverless project to AWS

### PR DESCRIPTION
**Pull request by mattcfox**
This is currently setup and tested with the Supply project. Further work
may be necessary to genericize for other projects.

### CODE REVIEW

**Review by zjromani — APPROVED** [review]
Nice job!

File: sls_deploy/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    shell: bash
+  - run: yarn prep-deploy
+    shell: bash
+  - run: yarn sls deploy --stage=$STAGE_NAME --region=us-east-1
+    shell: bash
+    env:
+      AWS_ACCESS_KEY_ID: ${{ inputs.aws_access_key_id }}
+      AWS_SECRET_ACCESS_KEY: ${{ inputs.aws_secret_access_key }}
+      STAGE_NAME: ${{ inputs.stage }}
+      S3_BUCKET: ${{ inputs.stage == 'production' && 'he-supply-production' || 'he-supply-stage' }}
```
Comment (dalexj) [inline sls_deploy/action.yml:35]: could you move `he-supply-xxx` out of this file and into inputs?

File: sls_deploy/action.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Deploy tag to AWS
+description: Deploys a given tag or most recent tag to heroku
```
Comment (sdrioux) [inline sls_deploy/action.yml:2]: I'm assuming this should say AWS - not Heroku?

File: sls_deploy/action.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+inputs:
+  aws_access_key_id:
+    description: AWS Access Key ID
+    required: true
+  aws_secret_access_key:
+    description: Secret credentials for AWS
+    required: true
```
Comment (sdrioux) [inline sls_deploy/action.yml:9]: Should these be secrets that are stored in Github instead?

Comment (mattcfox) [reply sls_deploy/action.yml:9]: These are, they are passed in from the parent workflow, there they are pulled out of that repository's secret store. I'm not sure if actions from outside of the repository can access the repository's secrets, that might only be in the parent workflow.

File: sls_deploy/action.yml (hunk 2 above)
Comment (mattcfox) [reply sls_deploy/action.yml:2]: Yes, I'll fix that

File: sls_deploy/action.yml (hunk 1 above)
Comment (mattcfox) [reply sls_deploy/action.yml:35]: For sure, that shouldn't be in here.

File: sls_deploy/action.yml (hunk 2 above)
Comment (mattcfox) [reply sls_deploy/action.yml]: Name and description updated

File: sls_deploy/action.yml (hunk 3 above)
Comment (zjromani) [reply sls_deploy/action.yml:9]: There is also the concept of [organization secrets](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-organization) that we could look at, but as Matt said, they are being passed from the `supply` repo for now.

**Review by sdrioux — APPROVED** [review]

**Pull request merged**